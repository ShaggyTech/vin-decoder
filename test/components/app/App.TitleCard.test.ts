// import flushPromises from 'flush-promises';

import { mount } from '@vue/test-utils';

import AppTitleCard from '@/components/app/TitleCard.vue';

const factory = (options: { [propName: string]: any } = {}) => {
  return mount(AppTitleCard, {
    ...options,
  });
};

describe('AppTitleCard Component Tests', () => {
  // let wrapper: ReturnType<typeof factory>;

  test('base component state, matches snapshot', () => {
    const wrapper = factory({
      sync: false,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  test('Props "header" and "subheader"', () => {
    const wrapper = factory({
      sync: false,
      propsData: {
        header: {
          innerText: 'This is a heading',
          class: 'some-class',
        },
        subHeader: {
          innerText: 'This is a sub-heading',
          class: 'some-other-class',
        },
      },
    });

    /* innerText and class are bound to id#Header element */
    expect(wrapper.find('#Header').text()).toBe('This is a heading');
    expect(wrapper.find('#Header').classes()).toContain('some-class');

    /* innerText and class are bound to id#SubHeader element  */
    expect(wrapper.find('#SubHeader').text()).toBe('This is a sub-heading');
    expect(wrapper.find('#SubHeader').classes()).toContain('some-other-class');

    expect(wrapper.element).toMatchSnapshot();
  });
});
