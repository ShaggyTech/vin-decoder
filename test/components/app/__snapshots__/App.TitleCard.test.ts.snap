// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitleCard Component Tests Props "header" and "subheader" 1`] = `
<div
  class="v-toolbar__title"
>
  <nuxt-link-stub
    to="/"
  >
    <span
      class="some-class"
      id="Header"
      innertext="This is a heading"
    >
      This is a heading
    </span>
     
    <span
      class="some-other-class"
      id="SubHeader"
      innertext="This is a sub-heading"
    >
      This is a sub-heading
    </span>
  </nuxt-link-stub>
</div>
`;

exports[`AppTitleCard Component Tests base component state, matches snapshot 1`] = `
<div
  class="v-toolbar__title"
>
  <nuxt-link-stub
    to="/"
  >
    <span
      class=""
      id="Header"
      innertext=""
    >
      
    </span>
     
    <span
      class=""
      id="SubHeader"
      innertext=""
    >
      
    </span>
  </nuxt-link-stub>
</div>
`;
